_stbt/core.py

```diff
@@ -933,9 +933,6 @@ def _match_all(self, image, frame, match_parameters, region):
 
         region = Region.intersect(_image_region(frame), region)
 
-        # Remove this copy once we've confirmed that doing so won't cause
-        # push_sample to doodle on the frame later.
-        frame = frame.copy()
         frame.flags.writeable = False
```

Comment:
Presumably we can remove this `frame.flags.writeable = False` too, because we already set it in `Display.on_new_sample`.

Reply:
Yes, good idea.  I see you've done this in a fixup.